doc/contribute/coding_guidelines_cpp.rst

```diff
@@ -0,0 +1,793 @@
+* **T002** Reserved names should not be used for preprocessor macros
+* **T004** Some keywords should be immediately followed by a colon (clang-format)
+* **T005** Keywords break and continue should be immediately followed by a semicolon (clang-format)
+* **T006** Keywords return and throw should be immediately followed by a semicolon or a single space (clang-format)
+* **T007** Semicolons should not be isolated by spaces or comments from the rest of the code (~ clang-format)
+* **T010** Identifiers should not be composed of 'l' and 'O' characters only
+* **T017** Unnamed namespaces are not allowed in header files
+
+Further transformations performed by clang-format
+*************************************************
+
+* Align trailing comments
+* Always break before multi-line strings
+* Always break template declarations
+* Break constructor initializers before comma
+* Pointer alignment: Left
+* Space before assignment operators
+* Spaces before trailing comments: 1
+* Spaces in parentheses
+* Spaces in square brackets
+
+Stopwatch example
+~~~~~~~~~~~~~~~~~
+
+For example, the ``stopwatch.h`` file could look like:
+
+.. code::
+
+   /*
+    *  stopwatch.h
+    *
+    *  This file is part of NEST.
+    *
+    *  Copyright (C) 2004 The NEST Initiative
+    *
+    *  NEST is free software: you can redistribute it and/or modify
+    *  it under the terms of the GNU General Public License as published by
+    *  the Free Software Foundation, either version 2 of the License, or
+    *  (at your option) any later version.
+    *
+    *  NEST is distributed in the hope that it will be useful,
+    *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+    *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    *  GNU General Public License for more details.
+    *
+    *  You should have received a copy of the GNU General Public License
+    *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+    *
+    */
+
+   #ifndef STOPWATCH_H
+   #define STOPWATCH_H
+
+   #include <sys/time.h>
+
+   #include <iostream>
+   #include <cassert>
+
+   namespace nest
+   {
+   class Stopwatch
+   {
+   public:
+     typedef size_t timestamp_t;
+     typedef size_t timeunit_t;
+
+     enum
+     {
+       MICROSEC = ( timeunit_t ) 1,
+       MILLISEC = MICROSEC * 1000,
+       SECONDS = MILLISEC * 1000,
+       MINUTES = SECONDS * 60,
+       HOURS = MINUTES * 60,
+       DAYS = HOURS * 24
+     };
+
+     Stopwatch();
+
+     void start();
+
+     void stop();
+
+     bool isRunning() const;
+
+     double elapsed( timeunit_t timeunit = SECONDS ) const;
+
+     timestamp_t elapsed_timestamp() const;
+
+     void reset();
+
+     void print( const char* msg = "",
+                 timeunit_t timeunit = SECONDS,
+                 std::ostream& os = std::cout ) const;
+
+     friend std::ostream& operator<<( std::ostream& os,
+                                      const Stopwatch& stopwatch );
```

Comment:
Please check if these are below 120 characters if the line-breaks are removed. To be sure, it is maybe best to copy the current version of the `Stopwatch` files from `libnestutil` to here.